attribute_set_completeness/models/attribute_set_owner_mixin.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class AttributeSetOwnerMixin(models.AbstractModel):
+
+    _inherit = "attribute.set.owner.mixin"
+
+    completion_rate = fields.Float(compute="_compute_completion_rate", readonly=True)
+    completion_state = fields.Selection(
+        selection=[("complete", "Complete"), ("not_complete", "Not complete")],
+        compute="_compute_completion_rate",
+        search="_search_complete_state",
+    )
+
+    @api.multi
+    def _compute_completion_rate(self):
+        for record in self:
+            attribute_set_id = record.attribute_set_id
+            completion_config = attribute_set_id.attribute_set_completeness_ids
+            if completion_config:
+                completion_rate = 0.0
+                for criteria in completion_config:
+                    field_name = criteria.field_id.name
+                    if record[field_name]:
+                        completion_rate += criteria.completion_rate
+                record.completion_rate = completion_rate
+                if completion_rate < 1:
```

Comment:
@Cedric-Pigeon There is no rule to enforce that the total theoretical completeness rate must be equal to 1. there is nothing to prevent defining on an attribute a theoretical completeness rate of 1000 and nothing to verify that the total of the theoretical rates is equal to 1. But maybe I'm missing something.

Reply:
just added a constrains